packages/core/scripts/admin-proposals/identifier.js

```diff
@@ -0,0 +1,278 @@
+    netId
+  );
+  await gasEstimator.update();
+  console.log(
+    `⛽️ Current fast gas price for Ethereum: ${web3.utils.fromWei(
+      gasEstimator.getCurrentFastPrice().toString(),
+      "gwei"
+    )} gwei`
+  );
+  const governor = new web3.eth.Contract(Governor.abi, _getContractAddressByName("Governor", netId));
+  const finder = new web3.eth.Contract(Finder.abi, _getContractAddressByName("Finder", netId));
+  const oracleAddress = await finder.methods
+    .getImplementationAddress(web3.utils.utf8ToHex(interfaceName.Oracle))
+    .call();
+  const oracle = new web3.eth.Contract(Voting.abi, oracleAddress);
+  const governorRootTunnel = new web3.eth.Contract(
+    GovernorRootTunnel.abi,
+    _getContractAddressByName("GovernorRootTunnel", netId)
+  );
+  if (polygonIdentifiers) {
+    console.group("\nℹ️  Relayer infrastructure for Polygon transactions:");
+    console.log(`- IdentifierWhitelist @ ${polygon_whitelist.options.address}`);
+    console.log(`- GovernorRootTunnel @ ${governorRootTunnel.options.address}`);
+    console.groupEnd();
+  }
+  console.group("\nℹ️  DVM infrastructure for Ethereum transactions:");
+  console.log(`- IdentifierWhitelist @ ${whitelist.options.address}`);
+  console.log(`- Finder @ ${finder.options.address}`);
+  console.log(`- Oracle @ ${oracle.options.address}`);
+  console.log(`- Governor @ ${governor.options.address}`);
+  console.groupEnd();
+
+  if (!verify) {
+    console.group("\n🌠 Proposing new Admin Proposal");
+
+    const adminProposalTransactions = [];
+    console.group("\n Key to understand the following logs:");
+    console.log(
+      "- 🟣 = Transactions to be submitted to the Polygon contracts are relayed via the GovernorRootTunnel on Etheruem. Look at this test for an example:"
+    );
+    console.log(
+      "    - https://github.com/UMAprotocol/protocol/blob/349401a869e89f9b5583d34c1f282407dca021ac/packages/core/test/polygon/e2e.js#L221"
+    );
+    console.log("- 🟢 = Transactions to be submitted directly to Ethereum contracts.");
+    console.groupEnd();
+
+    for (let i = 0; i < count; i++) {
+      if (identifiers && identifiers[i]) {
+        console.group(`\n🟢 Whitelisting identifier ${identifiers[i]} (UTF8: ${web3.utils.hexToUtf8(identifiers[i])})`);
+
+        // The proposal will only whitelist a new identifier if it isn't already whitelisted.
+        if (!(await whitelist.methods.isIdentifierSupported(identifiers[i]).call())) {
+          const addSupportedIdentifierData = whitelist.methods.addSupportedIdentifier(identifiers[i]).encodeABI();
+          console.log("- addSupportedIdentifierData", addSupportedIdentifierData);
+          adminProposalTransactions.push({ to: whitelist.options.address, value: 0, data: addSupportedIdentifierData });
+        } else {
+          console.log("- Identifier is already on whitelist. Nothing to do.");
+        }
+        console.groupEnd();
+      }
+
+      if (polygonIdentifiers && polygonIdentifiers[i]) {
```

Comment:
sheeesh. fair amount of complexity to deal with both kinds of proposals

Reply:
Yeap